clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -326,6 +340,7 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
             SelectionKey key = iterator.next();
             iterator.remove();
             KafkaChannel channel = channel(key);
+            long channelStartTimeNanos = recordTimePerConnection ? time.nanoseconds() : 0;
```

Comment:
Hmm, it seems a bit unfortunate that we need to do a `System.nanoTime` per selection key. Have we done any measurements on the overhead?